src/python/pants/engine/process_test.py

```diff
@@ -260,6 +263,18 @@ def test_create_files(rule_runner: RuleRunner) -> None:
     assert result.stdout == b"hellogoodbye"
 
 
+def test_process_output_symlink_aware(rule_runner: RuleRunner) -> None:
+    process = Process(
+        argv=("/bin/ln", "-s", "dest", "source"),
+        output_files=["source"],
```

Comment:
Having the `output_X` be `files`/`dirs` is a bit misleading now. It should be `output_globs`. But C'est la vie.

Reply:
REAPI v2.1 uses `output_paths with `output_files`, `output_directories`, and `output_symlinks` in the `ActionResult` protobuf message. If we rename, we should rename to that. See https://github.com/pantsbuild/pants/blob/6037e699b881011f845b5f7ad109fa8eae39cc3b/src/rust/engine/protos/protos/bazelbuild_remote-apis/build/bazel/remote/execution/v2/remote_execution.proto#L958.

Reply:
Not suggesting a rename, just pointing out the oddity.

Reply:
👍 

More mentioning because I am an advocate of us renaming at some point to remain consistent with REAPI (but not much of a priority to me).

Reply:
Specifically though I would advocate for `output_globs` so it's unambiguous that the paths are globs.

Reply:
Ah, care to enlighten me on how the `output_paths` work? Is it destination-aware (e.g. if the destination is a dir, captures the entire subdir, otherwise captures it as-is)?

Reply:
Exactly, the remote execution system is required to be aware of the type when capturing. If the `output_path` ends up being a directory, then the whole subtree is captured as a `Directory` proto and metadata is written to the `ActionResult.output_directories` field. Similarly files and symlinks are captured and metadata written to `output_files` and `output_symlinks` respectively. 

The `output_files` field on the Command/Action protos (which comes from Pants' `Process`) is deprecated but retains meaning of "an output path but also check that a file was actually captured and not a directory or symlink. Similar behavior for the other "what to capture" fields.

Reply:
For more info, see the link above to the REAPI spec and comments on `ActionResult` protobuf message.